### ISSUE TITLE
Omakub for the fresh Debian 12

### ISSUE THREAD

**Issue by puzanov** [basecamp/omakub#47]
I am a former Ubuntu user and switched to Debian two years ago and absolutely happy with its minimalistic approach and well tested apps. 

I know that Ubuntu is based on Debian. Will omakub run on fresh Debian 12? Or the regular debian tested repos are not fresh enought for the apps versions needed for omakub?

My feeling that if we want to help users create the best and beautifull Linux env for web develompent using Debian instead of Ubuntu probably will deliver much better stability and faster speed

**Comment by Kawsay** [comment]
> Will omakub run on fresh Debian 12

As it is right now it wont, but I think only few tweeks are required to make it work. 

For example, Debian does not support `add-apt-repository` out of the box (but it can be installed via the `software-properties-common` package)

**Comment by iljo-dp** [comment]
Currently it won't work on ubuntu as far as I know, and iirc this is primarly an ubuntu script as stated by dhh, so i'm not sure if support will ever be added for that

**Comment by con5ole** [comment]
I ran it on my Debian 12 and it mostly worked. It fails for a few apps but the basic setup seems fine. The biggest issue I've found so far was that ulauncher has to be installed differently on Debian vs Ubuntu but it was trivial to alter the omakub ulauncher script accordingly.  

```
sudo apt update && sudo apt install -y gnupg
gpg --keyserver keyserver.ubuntu.com --recv 0xfaf1020699503176
gpg --export 0xfaf1020699503176 | sudo tee /usr/share/keyrings/ulauncher-archive-keyring.gpg > /dev/null
echo "deb [signed-by=/usr/share/keyrings/ulauncher-archive-keyring.gpg] \
          http://ppa.launchpad.net/agornostal/ulauncher/ubuntu jammy main" \
          | sudo tee /etc/apt/sources.list.d/ulauncher-jammy.list
sudo apt update && sudo apt install ulauncher

# Start ulauncher to have it populate config before we overwrite
mkdir -p ~/.config/autostart/
cp ~/.local/share/omakub/configs/ulauncher.desktop ~/.config/autostart/ulauncher.desktop
gtk-launch ulauncher.desktop >/dev/null 2>&1
sleep 2 # ensure enough time for ulauncher to set defaults
cp ~/.local/share/omakub/configs/ulauncher.json ~/.config/ulauncher/settings.json

```

Seems to me that multiple distros could be added to Omakub simply by adding distro-specific "install"-subfolders and having the main setup script do a `cat /etc/*-release` to determine which subfolder to run the install-scripts from. 
There would be a lot of duplication but it would be easy to understand. 

Different desktop environments should perhaps be split off to separate branches though. "Omakub-KDE" etc.

**Comment by puzanov** [comment]
@con5ole Thanks for sharing. Did you install omakub on the fresh Debian 12? What type of ports did you use: stable or backports?

**Comment by con5ole** [comment]
> @con5ole Thanks for sharing. Did you install omakub on the fresh Debian 12? What type of ports did you use: stable or backports?

Stable, but it wasn't a fresh install. I started digging a little and actually found several things that didn't work properly out of the box, such as docker and ruby. 

The main issues for Debian seem to be 1) missing base dependencies such as a c compiler, zlib 2) old versions of some apps in the bookworm repos and 3) differences between Debian and Ubuntu for some apps. 

I made a fork (omakub_multidistro, not sure if it is polite to add a link here but it's not hard to find if you want to have a look) with fixes for most of the things I discovered. It's basically Omakub (as of yesterday) but with two paths, one for Debian and one for Ubuntu. I tested on both a clean ubuntu and clean debian 12 bookworm VM yesteday with good results.  

Remaining known issues for Debian are that Zellij is not added by default to Alacritty and neovim is a much older version that complains about the omakub config (but still runs). 
I also removed Typora, it didn't work immediately and seems to be commercial software so I didn't spend much time on it.
